.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,22 @@
+name: Build and Deploy
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+
+      - name: Install and Build 🔧
+        run: |
+          yarn
+          yarn build
```

Comment:
@JordanPisani `yarn build` might not work here... should probably use `yarn generate`